ts/input/tex/TexParser.ts

```diff
@@ -88,6 +88,7 @@ export default class TexParser {
     this.stack = new Stack(this.itemFactory, ENV, inner ? isInner : true);
     this.Parse();
     this.Push(this.itemFactory.create('stop'));
+    this.stack.env = ENV;
```

Comment:
> Is there the possibility that `stack._env` will be a different `EnvList` object during parsing than the original `ENV`?

I'm no longer sure why that change was made.  It looks like the only case where it would make a difference is if `ENV` were `null`, in which case `this.stack.env` would be the environment from the `StartItem`, and this would make it `null` instead.  That doesn't seem like a good idea.

It may be that this was something I tried to fix a problem, but forgot to remove when it didn't work.  I will remove it now, as I don't see how it would help.

Reply:
OOPS, I'm wrong, and I think I remember the reason now.  The `stack.env` may be changed after the parsing (in the `Push()` and `Pop()` methods of the stack), and this restores the original environment after the parsing is done, so that it contains whatever was in the global stack at one end of the parsing.  I think the `stop` item may be blowing away the stack environment and leaving it empty, so you can't find out what was defined in the outermost nesting level after the parsing is complete.

So I'm leaving it in after all.

Reply:
This was added during [5597660](https://github.com/mathjax/MathJax-src/commit/559766030acae651a1318dc3e1a24c206b23afa3), and I think it is this code that uses it:

https://github.com/mathjax/MathJax-src/blob/559766030acae651a1318dc3e1a24c206b23afa3/ts/input/tex/base/BaseMethods.ts#L756-L781

where `arg.stack.env` is used to look up the value set by `\hline`.

Reply:
I've just verified that this is, in fact, the case.